### PR TITLE
In erdpy, run ABI generator. Optimizations. Fixes. Github workflow for erdpy.

### PR DESCRIPTION
**Pull request by andreibancioiu**
- For Rust projects, run ABI generator upon building the WASM file.
 - For Rust projects, patch contract templates wrt. ABI module.
 - Cache templates repository (30 seconds). 
 - Add github workflow for erdpy.
 - Optimize running time for tests, build time (for rust projects).

### CODE REVIEW

**Review by dorin-iancu — APPROVED** [review]

File: erdpy/projects/templates.py [hunk 1]
```diff
@@ -161,6 +162,34 @@ def _patch_cargo_wasm(self):
...
+    def _patch_cargo_abi(self):
+        cargo_path = path.join(self.directory, "abi", TemplateRust.CARGO_TOML)
+        if not path.isfile(cargo_path):
+            return
+
+        cargo_file = CargoFile(cargo_path)
+        cargo_file.package_name = f"{self.project_name}-abi"
+        cargo_file.version = "0.0.1"
+        cargo_file.authors = ["you"]
+        cargo_file.edition = "2018"
```
Comment (andrei-marinica) [inline erdpy/projects/templates.py:174]: let's also add
publish = false

File: erdpy/projects/templates.py [hunk 2]
```diff
@@ -161,6 +162,34 @@ def _patch_cargo_wasm(self):
...
         )
 
+    def _patch_cargo_abi(self):
+        cargo_path = path.join(self.directory, "abi", TemplateRust.CARGO_TOML)
+        if not path.isfile(cargo_path):
+            return
+
+        cargo_file = CargoFile(cargo_path)
+        cargo_file.package_name = f"{self.project_name}-abi"
+        cargo_file.version = "0.0.1"
```
Comment (andrei-marinica) [inline erdpy/projects/templates.py:172]: I used 0.0.0 for everything in the template smart contracts, not sure what is the best way to do it though

**Review by andrei-marinica — APPROVED** [review]

File: erdpy/projects/templates.py (hunk 2 above)
Comment (andreibancioiu) [reply erdpy/projects/templates.py:172]: I'll fix this in https://github.com/ElrondNetwork/elrond-sdk/pull/142.

File: erdpy/projects/templates.py (hunk 1 above)
Comment (andreibancioiu) [reply erdpy/projects/templates.py:174]: I'll fix this in https://github.com/ElrondNetwork/elrond-sdk/pull/142.

**Pull request merged**